docs/source/torch.compiler_deepdive.rst

```diff
@@ -206,3 +206,53 @@ By generating this `resume_at` function, we force the remainder of the
 function to be executed in a new Python frame which recursively
 triggers TorchDynamo to restart its capture once execution reaches that
 point for the first time.
+
+How to inspect artifacts generated by TorchDynamo?
+----------------------------------------------
+
+To inspect the artifacts generated by TorchDynamo, there is an API `torch._dynamo.eval_frame._debug_get_cache_entry_list` that retrieves compiled code and guards out of a function's `__code__` object. A compiled function can have several cache entries, and each cache entry consists a generated function to check guards, and a `types.CodeType` object to keep the code to be executed if the guarding conditions are satisfied.
+
+.. code-block:: python
+
+   from torch._dynamo.eval_frame import _debug_get_cache_entry_list
+   cache_entries = _debug_get_cache_entry_list(toy_example._torchdynamo_orig_callable.__code__)
+   guard, code = cache_entries[0]
+   # the guard takes an input frame, and tells whether a re-compilation should be triggered.
+   import inspect
+   print(inspect.getfullargspec(guard))
+   # if you know python bytecode, you can understand the following code.
+   import dis
+   dis.dis(guard)
+   dis.dis(code)
+
+The compiled bytecode, printed by `dis.dis(code)`, will call the result of the backend compiler function which is stored inside a global variable such as `__compiled_fn_0` in the module containing the original function.
+
+The generated bytecodes are roughly equivalent to the following Python (converted manually for illustration purposes).
+
+.. code-block:: python
+
+   def compiled_example(a, b):
+       # behind the scene, pytorch C code checks the guarding condition
+       # if all guard fails, trigger re-compile
+       # else, run the compiled code
+       # after some setup work, the code finally looks like the following
+       x, _ = __compiled_fn_0._torchdynamo_orig_callable(a, b)
+       if not _:
```

Comment:
what's `_` could you name the variable?

Reply:
It's a temp variable to hold the result of `b.sum() < 0`. Fixed in [e3fc8a3](https://github.com/pytorch/pytorch/pull/106860/commits/e3fc8a3f426f9075404d9bd843d8edfc5c7b40d7).